src/Documents.jl

```diff
@@ -307,9 +298,28 @@ function Document(;
         [],
         [],
         Dict{Markdown.Link, String}(),
-        Set{Symbol}(),
+        Set{Symbol}()
     )
-    Document(user, internal)
+
+    plugin_dict = Dict{DataType, DocumenterPlugin}()
+    for plugin in plugins
+        plugin isa DocumenterPlugin ||
+            throw("$(typeof(plugin)) is not a subtype of `DocumenterPlugin`.")
+        haskey(plugin_dict, typeof(plugin)) &&
+            throw("only one copy of $(typeof(plugin)) may be passed.")
+        plugin_dict[typeof(plugin)] = plugin
+    end
+
+    Document(user, internal, plugin_dict)
+end
+
+function plugin_settings(doc::Document, type::DataType)
+    @assert type <: DocumenterPlugin
+    if !haskey(doc.plugins, type)
+        doc.plugins[type] = type()
```

Comment:
We should document that a `DocumenterPlugin` needs to have `T()` construct available.